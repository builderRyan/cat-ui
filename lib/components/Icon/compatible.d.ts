/**
* icon type 兼容处理
*
*/
export declare function compatible(type: any): string;
