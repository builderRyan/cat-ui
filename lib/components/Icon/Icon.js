"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_1 = require("react");
var classnames_1 = require("classnames");
var styled_1 = require("./styled");
var Icon = (function (_super) {
    __extends(Icon, _super);
    function Icon() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Icon.prototype.render = function () {
        var _a = this.props, className = _a.className, style = _a.style, type = _a.type;
        var classes = classnames_1.default('hmly-icon', "hmly-icon-" + type, className);
        return (React.createElement(React.Fragment, null,
            React.createElement("i", { className: classes, style: style }),
            React.createElement(styled_1.StyledIcon, null)));
    };
    return Icon;
}(react_1.PureComponent));
exports.default = Icon;
//# sourceMappingURL=Icon.js.map