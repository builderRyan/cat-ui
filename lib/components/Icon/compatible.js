"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var IconTypeEnum_1 = require("./IconTypeEnum");
/** 老type映射成新的 */
var TypeMap;
(function (TypeMap) {
    TypeMap["question-circle"] = "question";
    TypeMap["share3"] = "share";
    TypeMap["share1"] = "share";
    TypeMap["share2"] = "share";
    TypeMap["ins"] = "instagram";
    TypeMap["drag1"] = "drag-mouse";
    TypeMap["drag2"] = "drag";
    TypeMap["prev"] = "left";
    TypeMap["next"] = "right";
    TypeMap["eye-close"] = "invisible";
    TypeMap["eye-open"] = "visible";
})(TypeMap || (TypeMap = {}));
/**
* icon type 兼容处理
*
*/
function compatible(type) {
    return IconTypeEnum_1.IconType[type] || TypeMap[type];
}
exports.compatible = compatible;
//# sourceMappingURL=compatible.js.map