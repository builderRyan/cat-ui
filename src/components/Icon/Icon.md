new Examplate 


```jsx
let NewIcon = () =>{
  return (
    <div className="icon-box">
      
<div className='icon-item'>
  <Icon type={Icon.IconType.close} />
  close
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.down} />
  down
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.right} />
  right
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.left} />
  left
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.up} />
  up
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.instagram} />
  instagram
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.twitter} />
  twitter
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.facebook} />
  facebook
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.locked} />
  locked
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.unlocked} />
  unlocked
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.volumeLow} />
  volume-low
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.volumeHigh} />
  volume-high
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.volumeMuted} />
  volume-muted
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.loopList} />
  loop-list
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.loopSingle} />
  loop-single
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.shuffle} />
  shuffle
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.question} />
  question
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.pause} />
  pause
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.playerNext} />
  player-next
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.play} />
  play
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.playerPrev} />
  player-prev
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.sortDown} />
  sort-down
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.playlist} />
  playlist
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.sortUp} />
  sort-up
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.share} />
  share
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.liked} />
  liked
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.like} />
  like
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.visible} />
  visible
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.invisible} />
  invisible
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.drag} />
  drag
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.dragMouse} />
  drag-mouse
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.upload} />
  upload
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.enter} />
  enter
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.edit} />
  edit
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.download} />
  download
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.stats} />
  stats
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.delete} />
  delete
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.more} />
  more
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.filter} />
  filter
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.people} />
  people
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.disc} />
  disc
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.forward} />
  forward
</div>

<div className='icon-item'>
  <Icon type={Icon.IconType.backward} />
  backward
</div>

      <style>
        {`
          .icon-box .hmly-icon {
            margin-right: 10px;
          }
          .icon-box .icon-item {
            display: inline-block;
            width: 20%;
            margin-bottom: 20px;
          }
        `}
      </style>
    </div>
  )
}
<NewIcon />
```

   
 
Example

```jsx
class Example extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    console.log('Example:handleClick')
  }

  render () {
    return (<div className='icon-box'>
      <div className='icon-item'>
        <Icon type='question-circle' onClick={this.handleClick} />
        question-circle
      </div>
      <div className='icon-item'>
        <Icon type='question' />
        question
      </div>
      <div className='icon-item'>
        <Icon type='share3' />
        share3
      </div>
      <div className='icon-item'>
        <Icon type='share2' />
        share2
      </div>
      <div className='icon-item'>
        <Icon type='share1' />
        share1
      </div>
      <div className='icon-item'>
        <Icon type='more' />
        more
      </div>
      <div className='icon-item'>
        <Icon type='stats' />
        stats
      </div>
      <div className='icon-item'>
        <Icon type='forward' />
        forward
      </div>
      <div className='icon-item'>
        <Icon type='backward' />
        backward
      </div>
      <div className='icon-item'>
        <Icon type='ins' />
        ins
      </div>
      <div className='icon-item'>
        <Icon type='twitter' />
        twitter
      </div>
      <div className='icon-item'>
        <Icon type='facebook' />
        facebook
      </div>
      <div className='icon-item'>
        <Icon type='crown' />
        crown
      </div>
      <div className='icon-item'>
        <Icon type='early-access' />
        early-access
      </div>
      <div className='icon-item'>
        <Icon type='hi-circle' />
        hi-circle
      </div>
      <div className='icon-item'>
        <Icon type='ad-free' />
        ad-free
      </div>
      <div className='icon-item'>
        <Icon type='close' />
        close
      </div>
      <div className='icon-item'>
        <Icon type='upload' />
        upload
      </div>
      <div className='icon-item'>
        <Icon type='delete' />
        delete
      </div>
      <div className='icon-item'>
        <Icon type='edit' />
        edit
      </div>
      <div className='icon-item'>
        <Icon type='up' />
        up
      </div>
      <div className='icon-item'>
        <Icon type='download' />
        download
      </div>
      <div className='icon-item'>
        <Icon type='process' />
        process
      </div>
      <div className='icon-item'>
        <Icon type='filter' />
        filter
      </div>
      <div className='icon-item'>
        <Icon type='enter' />
        enter
      </div>
      <div className='icon-item'>
        <Icon type='drag1' />
        drag1
      </div>
      <div className='icon-item'>
        <Icon type='drag2' />
        drag2
      </div>
      <div className='icon-item'>
        <Icon type='prev' />
        prev
      </div>
      <div className='icon-item'>
        <Icon type='next' />
        next
      </div>
      <div className='icon-item'>
        <Icon type='eye-open' />
        eye-open
      </div>
      <div className='icon-item'>
        <Icon type='eye-close' />
        eye-close
      </div>
      <style>
        {`
          .icon-box .hmly-icon {
            margin-right: 10px;
          }
          .icon-box .icon-item {
            display: inline-block;
            width: 20%;
            margin-bottom: 20px;
          }
        `}
      </style>
    </div>
    )
  }
}

<Example />
```

