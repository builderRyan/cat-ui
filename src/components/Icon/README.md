# Icon 字体图标

## Example

```jsx
<Icon type='close' />
```

## API

|   属性    |     类型     | 默认值 |   描述   | 是否必填 |
|:---------:|:------------:|:------:|:--------:|:--------:|
| className |   `string`   |   -    |   类名   |    -     |
|   style   |   `object`   |   -    | 行内样式 |    -     |
|   type    |   `string`   |   -    |   类型   |    是    |
|  onClick  | `() => void` |   -    | 点击回调 |    -     |

