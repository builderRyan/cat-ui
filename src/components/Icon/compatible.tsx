
import {IconType} from './IconTypeEnum'

/** 老type映射成新的 */
 enum TypeMap {
    "question-circle"= "question",
    "share3"= "share",
    "share1"= "share",
    "share2"= "share",
    "ins"= "instagram",
    "drag1"="drag-mouse",
    "drag2"="drag",
    "prev"="left",
    "next"="right",
    "eye-close"="invisible",
    "eye-open"="visible"
 }

 /**
 * icon type 兼容处理
 *  
 */
 export function compatible (type: any) {
    return TypeMap[type as keyof typeof TypeMap] || type;
 }
 


 
 
 