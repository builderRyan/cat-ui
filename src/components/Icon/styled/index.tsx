import { createGlobalStyle } from 'styled-components'
import { primaryBtnColor } from '../../../styles/theme'

const StyledIcon = createGlobalStyle`
  @font-face {
    font-family: 'Himalaya';  /* project id 1121757 */
    src: url('//at.alicdn.com/t/font_1121757_i1uxm83odrh.eot');
    src: url('//at.alicdn.com/t/font_1121757_i1uxm83odrh.eot?#iefix') format('embedded-opentype'),
    url('//at.alicdn.com/t/font_1121757_i1uxm83odrh.woff2') format('woff2'),
    url('//at.alicdn.com/t/font_1121757_i1uxm83odrh.woff') format('woff'),
    url('//at.alicdn.com/t/font_1121757_i1uxm83odrh.ttf') format('truetype'),
    url('//at.alicdn.com/t/font_1121757_i1uxm83odrh.svg#Himalaya') format('svg');
  }

  .hmly-icon {
    font-family: "Himalaya" !important;
    font-size: 16px;
    font-style: normal;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  .hmly-icon-question-circle:before {
    content: "\\e620";
  }

  .hmly-icon-question:before {
    content: "\\e61f";
  }

  .hmly-icon-share3:before {
    content: "\\e61e";
  }

  .hmly-icon-share2:before {
    content: "\\e61d";
  }

  .hmly-icon-share1:before {
    content: "\\e61c";
  }

  .hmly-icon-more:before {
    content: "\\e621";
  }

  .hmly-icon-stats:before {
    content: "\\e61a";
  }

  .hmly-icon-forward:before {
    content: "\\e619";
  }

  .hmly-icon-backward:before {
    content: "\\e618";
  }

  .hmly-icon-ins:before {
    content: "\\e617";
  }

  .hmly-icon-twitter:before {
    content: "\\e616";
  }

  .hmly-icon-facebook:before {
    content: "\\e615";
  }

  .hmly-icon-crown:before {
    content: "\\e614";
  }

  .hmly-icon-early-access:before {
    content: "\\e613";
  }

  .hmly-icon-hi-circle:before {
    content: "\\e612";
  }

  .hmly-icon-ad-free:before {
    content: "\\e611";
  }

  .hmly-icon-close:before {
    content: "\\e610";
  }

  .hmly-icon-upload:before {
    content: "\\e627";
  }

  .hmly-icon-delete:before {
    content: "\\e626";
  }

  .hmly-icon-edit:before {
    content: "\\e625";
  }

  .hmly-icon-up:before {
    content: "\\e624";
  }

  .hmly-icon-download:before {
    content: "\\e623";
  }

  .hmly-icon-process:before {
    content: "\\e622";
  }

  .hmly-icon-filter:before {
    content: "\\e62a";
  }

  .hmly-icon-enter:before {
    content: "\\e62b";
  }

  .hmly-icon-drag1:before {
    content: "\\e629";
  }

  .hmly-icon-drag2:before {
    content: "\\e628";
  }
`;

export { StyledIcon }
