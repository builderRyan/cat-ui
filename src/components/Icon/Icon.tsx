import * as React from 'react'
import { PureComponent } from 'react'
import classNames from 'classnames'
import { StyledIcon } from './styled'

export interface IconProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** 图标类型 */
  type: string;
}

/**
 * 图标
 */
class Icon extends PureComponent<IconProps, any> {

  render() {
    const { className, style, type } = this.props
    const classes = classNames('hmly-icon', `hmly-icon-${type}`, className)

    return (<React.Fragment>
      <i
        className={classes}
        style={style} />
      <StyledIcon />
    </React.Fragment>
    )
  }
}

export default Icon
