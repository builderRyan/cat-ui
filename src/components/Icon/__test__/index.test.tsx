import * as React from 'react'
import * as Enzyme from 'enzyme'
import { render } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import Icon from '../index'

Enzyme.configure({ adapter: new Adapter() })

describe('Icon', () => {
  test('renders correctly', () => {
    const wrapper = render(
      <Icon type='close' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('two plus two is four', () => {
    expect(2 + 2).toBe(4)
  })
})
