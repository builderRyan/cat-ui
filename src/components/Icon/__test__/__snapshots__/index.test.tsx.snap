// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon props: type 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-question",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 2`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-question",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 3`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-share",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 4`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-share",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 5`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-share",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 6`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-more",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 7`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-stats",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 8`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-forward",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 9`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-backward",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 10`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-instagram",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 11`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-twitter",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 12`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-facebook",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 13`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-crown",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 14`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-early-access",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 15`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-hi-circle",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 16`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-ad-free",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon props: type 17`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-close",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Icon renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "cat-icon cat-icon-close",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
